### ISSUE TITLE
Some problems about compiling environment

### ISSUE THREAD

**Issue by NHW2017** [YoungXIAO13/FewShotDetection#14]
Hello! I have some problems when running the program. I suspect that it is caused by a problem with my configuration environment, so I want to ask about your configuration environment, including the version of ubuntu, the version of cuda and the model of the graphics card

**Comment by YoungXIAO13** [comment]
Hi @NHW2017 

Yes, the environment configuration is important, especially use the correct PyTorch version and CUDA version.

Two commands might be useful when installing the environments:
```
conda install pytorch=0.4.1 cuda80 -c pytorch
conda install -c anaconda cudatoolkit=8.0
```

You probably need to compile everything with **CUDA 8.0** and it should be ok for any ubuntu version that supports it.
In my case, I've tested in both ubuntu 14.04 or 16.04 using Titan X (12G), I also tested with PyTorch 0.4.0 and 0.4.1, both works fine.
However, I've never tried with CUDA 10 and any version of PyTorch >= 1.0

**Comment by NHW2017** [comment]
> Hi @NHW2017
> 
> Yes, the environment configuration is important, especially use the correct PyTorch version and CUDA version.
> 
> Two commands might be useful when installing the environments:
> 
> ```
> conda install pytorch=0.4.1 cuda80 -c pytorch
> conda install -c anaconda cudatoolkit=8.0
> ```
> 
> You probably need to compile everything with **CUDA 8.0** and it should be ok for any ubuntu version that supports it.
> In my case, I've tested in both ubuntu 14.04 or 16.04 using Titan X (12G), I also tested with PyTorch 0.4.0 and 0.4.1, both works fine.
> However, I've never tried with CUDA 10 and any version of PyTorch >= 1.0

thank you for your reply! I have another question. I used the same hardware and environment configuration as yours. It takes about 100 minutes to train an EPOCH on the VOC data set. It takes about 35 hours to train 21 epochs at phase=1, which is too long. So I want to ask you if it takes such a long time under the same conditions. If not, it means that there is a problem with my environment.

**Comment by YoungXIAO13** [comment]
Actually, the training time is correct. The base class training on VOC also took me around 30 hours.
To accelerate, one solution would be implementing parallel training with multiple GPUs, which I haven't tried yet.

**Comment by NHW2017** [comment]
> Actually, the training time is correct. The base class training on VOC also took me around 30 hours.
> To accelerate, one solution would be implementing parallel training with multiple GPUs, which I haven't tried yet.

 Thanks. How about COCO? How long does it take to train Coco once

**Comment by YoungXIAO13** [comment]
The base training on COCO would take ~10 days, which is unfortunately too long with only one GPU of 12G.
A work around is to increase the batch size and reduce the training epochs linearly, say 4 times less epochs if you have a GPU of 48G.

**Comment by NHW2017** [comment]
> The base training on COCO would take ~10 days, which is unfortunately too long with only one GPU of 12G.
> A work around is to increase the batch size and reduce the training epochs linearly, say 4 times less epochs if you have a GPU of 48G.

Sorry to interrupt, I would like to ask if you have tried adding Focal loss or other loss function with similar functions to the code? How's the effect?  Looking forward to you reply ,thanks!

**Comment by YoungXIAO13** [comment]
Hi @NHW2017 

I've not tried other loss functions and have no plan to try in near future. It would be interesting to study the effect.
